### PR TITLE
Re-implement Editbox on Android platform.

### PR DESCRIPTION
**Pull request by zilongshanren**
This PR fixed issue: 
https://github.com/cocos2d/cocos2d-x/issues/13147

This PR also fixed issue:

https://github.com/cocos2d/cocos2d-x/issues/9572

For the new EditBox:
1. Added a ResizeLayout for updating layout when soft keyboard is shown.
   This is because some Android devices won't refresh the view and when
   the view is panned, the native EditText will turn black. This is really
   a hack, but it works for many Android devices from 2.3 to 5.0.
   When keyboard is beginning to show, the ResizeLayout's doLayout
   method will be called 24 frames per second. Once the keyboard is closed,
   or the user begins to input, the doLayout method will be disabled.
2. EditText is working in a different UI thread, so all the data
   are passed to GL thread via callbacks in UI thead.
3. It also overrides onKeyDown method in Cocos2dxEditBox.
   When the EditText got the focus, users could press back key to exit
   the game. This behavior is not allowed. We give developer the choice to
   handle back key event manually.
4. It also add `static Rect convertBoundingBoxToScreen(Node* node);` in ui::Helper class to remove redundant code. And this function is very useful for synchronizing native control transform with OpenGL transform.

### CODE REVIEW

**Comment by zilongshanren** [comment]
@pandamicro  @fusijie  @WenhaiLin  
please help to review this PR, thanks.

**Comment by pandamicro** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
@pandamicro  Could this PR be merged now?

**Comment by pandamicro** [comment]
Yes, I think it should be merged so that QA team can test it in the next week

**Comment by pandamicro** [comment]
Besides, it should be link to an issue

**Comment by zilongshanren** [comment]
It fixes issue https://github.com/cocos2d/cocos2d-x/issues/13147

**Comment by zilongshanren** [comment]
@pandamicro  please merge...

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxEditBoxHelper.java [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    }
+
+    public static void setVisible(final int index, final boolean visible) {
+        mCocos2dxActivity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                Cocos2dxEditBox editBox = mEditBoxArray.get(index);
+                if (editBox != null) {
+                    editBox.setVisibility(visible ? View.VISIBLE : View.GONE);
+                    if (visible) {
```
Comment (fusijie) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxEditBoxHelper.java:274]: why not?

``` java
if(_oldVisible != visible)
{
    ...do sth.
}
```

The same to other setBool.

**Pull request merged**